Adding documentation for append only indices (#9809)
opensearch-project/documentation-website: _install-and-configure/configuring-opensearch/index-settings.md

- `index.use_compound_file` (Boolean): This setting controls the Apache Lucene `useCompoundFile` index writer settings, which specifies whether newly written segment files will be packed into a compound file. Default is `true`.

> - `index.append_only.enabled` (Boolean): This setting enabled for indices on which we want to prevent any forms of updates. Default is `false`.

Reviewer: ```suggestion
- `index.append_only.enabled` (Boolean): This setting prevents any index updates. Default is `false`.
```

> - `index.append_only.enabled` (Boolean): This setting enabled for indices on which we want to prevent any forms of updates. Default is `false`.

Reviewer 2: ```suggestion
- `index.append_only.enabled` (Boolean): Set to `true` to prevent any updates to documents in the index. Default is `false`.
```
